sdk/examples/intkey_cxx/int_key_transaction_processor.cpp

```diff
@@ -0,0 +1,292 @@
+ limitations under the License.
+------------------------------------------------------------------------------*/
+#include <iostream>
+#include <string>
+
+#include "log4cxx/logger.h"
+#include "log4cxx/basicconfigurator.h"
+
+#include "sawtooth/address_mapper.h"
+#include "sawtooth/transaction_processor.h"
+#include "json.hpp"
+
+using namespace log4cxx;
+
+using namespace nlohmann;
+
+static log4cxx::LoggerPtr logger(log4cxx::Logger::getLogger
+    ("sawtooth.IntKeyCXX"));
+
+static const std::string INTKEY_NAMESPACE = "intkey";
+
+// Custom Exception type defined to indicate any
+// failure in process a transaction.
+class InvalidTransaction: public std::runtime_error {
+ public:
+    explicit InvalidTransaction(std::string const& error)
+        : std::runtime_error(error)
+    {}
+};
+
+
+// utility function to provide copy conversion from vector of bytes
+// to a stl string container.
+void ToString(std::string* out, const std::vector<std::uint8_t>& in) {
+    out->resize(in.size());
+    const char* data = reinterpret_cast<const char*>(&in[0]);
+    out->assign(data, data+in.size());
+}
+
+// utility function to provide copy conversion from stl string container
+// to a vector of bytes.
+void ToVector(std::vector<std::uint8_t>* out, const std::string& in) {
+    out->resize(in.length());
+    memcpy(&((*out)[0]), in.data(), in.length());
+}
+
+// Handles the processing of a IntKey transactions.
+class IntKeyApplicator:  public sawtooth::TransactionApplicator {
+ public:
+    IntKeyApplicator(sawtooth::TransactionUPtr txn,
+        sawtooth::GlobalStateUPtr state) :
+        TransactionApplicator(std::move(txn), std::move(state)),
+            address_mapper(INTKEY_NAMESPACE) { }
+
+    void Apply() {
+        LOG4CXX_DEBUG(logger, "IntKeyApplicator::Apply");
+
+        const std::string& raw_data = this->txn->payload();
+        std::vector<uint8_t> data_vector(raw_data.begin(), raw_data.end());
+        json intkey_cmd = json::from_cbor(data_vector);
+
+        if (!intkey_cmd.is_object()) {
+            throw InvalidTransaction("CBOR Object as the encoded command");
+        }
+        auto verb_it = intkey_cmd.find("Verb");
+        if (verb_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Verb is required");
+        }
+        std::string verb = *verb_it;
+
+        auto name_it = intkey_cmd.find("Name");
+        if (name_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Name is required");
+        }
+        std::string name = *name_it;
+        if (name.length() == 0 || name.length() >= 20) {
+            throw InvalidTransaction("Name is invalid, name muste be between" \
+                " 1 and 20 characters in length");
+        }
+
+        auto value_it = intkey_cmd.find("Value");
+        if (value_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Value is required");
+        }
+        int value = (*value_it).get<uint32_t>();
```

Comment:
Is there value in using specific-sized integers - i.e., `int32_t`?  If for some reason the `int` is really 64-bit, then there is a potential glitch if you expect a negative number to be a "big" `uint32_t` and converting to `int` you expect it to become negative number.